pennylane/devices/default_qubit.py

```diff
@@ -185,6 +185,21 @@ def __init__(
             "Toffoli": self._apply_toffoli,
         }
 
+    @property
+    def stopping_condition(self):
+        def accepts_obj(obj):
+            if getattr(obj, "has_matrix", False):
```

Comment:
backprop and adjoint aren't quite working with power operators.

Reply:
Yes.

As a hotfix, we simply don't support `Pow` operators that are trainable.

I tried to make it so that `Pow` was only unsupported if we were on a backprogation device, but that would have been an immensely complicated fix.

When the `QNode` substitutes in a backpropagation device, the new device keeps the original device's `expand_fn` which communicates with the original device's `stopping_condition`. When on a backpropgation device that has been substituted in, the `stopping_condition` has no idea that it is on a backpropagation device.